docs/console/incremental-guide.md

```diff
@@ -0,0 +1,586 @@
+# Writing Console Apps: A Step by Step Guide
+
+This guide will show you how to use the .NET CLI tooling to build cross-platform console apps incrementally.  It will start with the most basic console app and eventually span multiple projects, including testing.
+
+If you're unfamiliar with the .NET CLI toolset, read [the overview](overview.md).  It's also helpful to have an understanding of the [console app paradigm](paradigm.md) and how [native binary compilation](single-binaries.md) works.
```

Comment:
I think that it's a little hard to describe because there are *so many* types of things you can write which are technically console apps.  The idea of a "Console App" is often conflated with the "Simple, on-off Console App", when that's definitely not the case.  A microservice running in a container is just as much a console app as a simple utility.  I agree with @blackdwarf that there is a sort of "paradigm" here that we could establish, with best practices highlighting that.